src/main/scala/com/datastax/gatling/plugin/checks/DseCheckSupport.scala

```diff
@@ -6,49 +6,9 @@
 
 package com.datastax.gatling.plugin.checks
 
-import io.gatling.core.session.ExpressionSuccessWrapper
-
 trait DseCheckSupport {
 
-  // start global checks
-  lazy val exhausted = GenericChecks.exhausted
-  lazy val applied = GenericChecks.applied
-  lazy val rowCount = GenericChecks.rowCount
-
-  // execution info and subsets
-  lazy val executionInfo = GenericChecks.executionInfo
-  lazy val achievedCL = GenericChecks.achievedConsistencyLevel
-  lazy val pagingState = GenericChecks.pagingState
-  lazy val queriedHost = GenericChecks.queriedHost
-  lazy val schemaAgreement = GenericChecks.schemaInAgreement
-  lazy val successfulExecutionIndex = GenericChecks.successfulExecutionIndex
-  lazy val triedHosts = GenericChecks.triedHosts
-  lazy val warnings = GenericChecks.warnings
-
-  // start CQL only checks
   lazy val resultSet = CqlChecks.resultSet
-  lazy val allRows = CqlChecks.allRows
-  lazy val oneRow = CqlChecks.oneRow
-
-  // start Graph only checks
-  lazy val graphResultSet = GraphChecks.graphResultSet
-  lazy val allNodes = GraphChecks.allNodes
-  lazy val oneNode = GraphChecks.oneNode
-
-  def edges(columnName: String) = GraphChecks.edges(columnName)
-
-  def vertexes(columnName: String) = GraphChecks.vertexes(columnName)
-
-  def paths(columnName: String) = GraphChecks.paths(columnName)
-
-  def properties(columnName: String) = GraphChecks.paths(columnName)
-
-  def vertexProperties(columnName: String) = GraphChecks.vertexProperties(columnName)
-
-  /**
-    * Get a column by name returned by the CQL statement.
-    * Note that this statement implicitly fetches <b>all</b> rows from the result set!
-    */
-  def columnValue(columnName: String) = CqlChecks.columnValue(columnName.expressionSuccess)
```

Comment:
Removal of the checks above reflects an API change.  Going forward the intent is to access the underlying ResultSet impl directly (via the appropriate "resultSet" check) and then extract the desired information via .transform() + fns defined somewhere in the code.  Users can define these transform fns in their own simulations, in library code, etc.  This has the effect of making the check API simpler and more streamlined while allowing users to extend it in a natural way by defining their own fns.

Reply:
I agree with the simplification goal. And it is a good time to do this, given that all user simulations will break anyway, because of the DseCluster/DseSession class change in the 4.x branch. May I ask for a small documentation where this decision is explained and the replacement code is provided for deleted checks, to make things easier for users?

Reply:
Extensive scaladoc has been added on this point in DseCheckSupport.  @pingtimeout would you mind taking a look and confirming that this addresses the points you wanted to see covered?